build.gradle

```diff
@@ -60,3 +61,27 @@ jacocoTestReport {
 }
 
 build.dependsOn jacocoTestReport
+
+pmd {
+    toolVersion = '5.4.1'
+    ignoreFailures = false
+    ruleSetFiles = files('./code_quality_tools/pmd.xml')
+}
+
+task('pmd', type: Pmd, dependsOn: 'assemble') {
+    description = 'Run pmd'
+    group = 'verification'
+
+    ruleSets = []
+
+    source = fileTree('src/main/java')
+    include '**/*.java'
+    exclude '**/gen/**'
+
+    reports {
+        html.enabled = true
+        xml.enabled = true
+    }
+}
+
+build.dependsOn 'pmd'
```

Comment:
nit: put newline at the end

<img width="266" alt="screen shot 2016-06-22 at 1 10 26 pm" src="https://cloud.githubusercontent.com/assets/967132/16262995/bea00f42-387a-11e6-961b-18dfd99f4926.png">